examples/nlp/language_modeling/megatron_change_num_partitions.py

```diff
@@ -900,7 +903,31 @@ def main():
     if args.model_file is None and args.model_extracted_dir is None:
         raise ValueError("Cannot pass model_file and model_extracted_dir as None at the same time.")
 
-    trainer = Trainer(devices=1, strategy=NLPDDPStrategy(), accelerator="cpu", precision=precision)
+    tmp_cfg = cls.restore_from(
+        restore_path=args.model_file,
+        trainer=Trainer(devices=1, strategy=NLPDDPStrategy(), accelerator="cpu", precision=precision),
+        map_location=torch.device("cpu"),
+        return_config=True,
+    )
+    plugins = []
+    if precision in [16, '16', 'bf16', '16-mixed', 'bf16-mixed']:
+        scaler = None
+        if precision in [16, '16', '16-mixed']:
+            scaler = GradScaler(
```

Comment:
why do we need to setup the GradScaler in this script?

Reply:
https://github.com/NVIDIA/NeMo/blob/fb87053f62780256029632494228389396ce1fd9/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L774

this line requires scalar if precision is fp16.

Reply:
this is just a TP/PP conversion script, I am not sure what needs a GradScaler here

Reply:
the code I point to runs when initializing the model. Thus it will error if user set precision to 16 when doing the conversion without a scalar.

Reply:
ugh... I see, looks like this moved from fwd_bwd to model init with config update
cool, can fix it outside this PR